macros.in

```diff
@@ -1104,6 +1104,24 @@ package or when debugging this package.\
 # arch macro for 64-bit LOONGARCH processors
 %loongarch64	loongarch64
 
+#------------------------------------------------------------------------
+# Use in %install to generate fully-quoted file lists. For example,
+#
+# %install
+# ...
+# %collect [-r <root>] %{name}
+# ...
+# %files -f %{name}.list
+#
+%collect(r:)\
+rm -f %{1}.list \
+find %{buildroot}%{_prefix}/%{-r*} -type f -print | \\\
+perl -pe 's!%{buildroot}!!g;'			    \\\
```

Comment:
Ugh, let's not drag Perl into any stock rpm macros, we don't want any new dependencies on the gorilla which we're trying to walk out the door.

Also I don't recall ever coming across a case that would've needed this thing from the dusty 2001 doc, with the weird `%{_prefix}` and all. Which is not to say such cases don't exist, but whether it's common enough to warrant a macro I dunno.

An alternative, probably more robust way to achieve this would be adding a new "raw manifest" flag to `%files` which does not expand lines at all.

Reply:
Granted, this crossed my mind but I didn't check if we used perl already. Could be done with sed just as well, but otherwise yeah, you're probably right, just because it's documented as a possible use case, the actual usefulness would be questionable. One real-life use case would be the one described in #1749 (or the associated [email thread](http://lists.rpm.org/pipermail/rpm-list/2021-June/002048.html)) perhaps, but it's still not crucial to have in core macros for sure.

The raw manifest flag would indeed be a cleaner solution. I'd not go that far right now, though, as I agree it's not a feature that would be specifically requested by anyone, really.

So I'll just close this PR then. Thanks for the quick feedback! :smile: